src/main/webapp/scripts/editCard.js

```diff
@@ -0,0 +1,95 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the 'License');
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an 'AS IS' BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Query salecard from servlet and insert values in the form
+ * TODO: fetch from servlet
+ * TODO: handle case where a vendor does not have a card registered
+ */
+const querySalecard = () => {
+  let salecard = vendor.saleCard;
+
+  document.getElementById('business-name').value = salecard.businessName;
+  document.getElementById('business-description').value = salecard.description;
+  document.getElementById('business-startTime').value = salecard.startTime;
+  document.getElementById('business-endTime').value = salecard.endTime;
+  document.getElementById('delivery').checked = salecard.hasDelivery;
+  document.getElementById('business-distanceOfDelivery').value = salecard.location.radius;
+  document.getElementById('photo-description').value = salecard.picture.altText;
+  document.getElementById('business-lat').value = salecard.location.salePoint.latitude;
+  document.getElementById('business-lng').value = salecard.location.salePoint.longitude;
+  
+  // After the card is inserted on the body we can init the map
+  initMap();
+};
+
+/**
+ * Update location to inserts current vendor's location
+ */
+const updateLocation = () => {
+  // Get client current location
+  if (navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition((position) => {
+      document.getElementById('business-lat').value = position.coords.latitude;
+      document.getElementById('business-lng').value = position.coords.longitude;
+      
+      // Refresh current map
+      drawMap();
+    },
+      (error) => {
+        if (error.code == error.PERMISSION_DENIED)
```

Comment:
might want to catch more types of errors? or is permission denied the only possible one here.

Reply:
Thank you for that, I found that if I print `error.message` basically describes the error. I'll use that instead of:
`if (error.code == error.PERMISSION_DENIED)`